File: Codes/0680-valid-palindrome-ii.go
/*
 * @Author: LetMeFly
 * @Date: 2025-02-03 08:57:46
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-02-03 09:05:54
 */
package main

func isOk_VP(s string, l, r int) bool {
    for ; l < r; l, r = l + 1, r - 1 {
        if s[l] != s[r] {
            return false
        }
    }
    return true
}

func validPalindrome(s string) bool {
    for l, r := 0, len(s) - 1; l < r; l, r = l + 1, r - 1 {
        if s[l] != s[r] {
            return isOk_VP(s, l, r - 1) || isOk_VP(s, l + 1, r)
        }
    }
    return true
}

File: Codes/0680-valid-palindrome-ii.cpp
/*
 * @Author: LetMeFly
 * @Date: 2025-02-03 08:52:33
 * @LastEditors: LetMeFly.xyz
 * @LastEditTime: 2025-02-03 08:57:47
 */
#ifdef _WIN32
#include "_[1,2]toVector.h"
#endif

class Solution {
private:
    bool isOk(string& s, int l, int r) {
        for (; l < r; l++, r--) {
            if (s[l] != s[r]) {
                return false;
            }
        }
        return true;
    }
public:
    bool validPalindrome(string& s) {
        for (int i = 0, j = s.size() - 1; i < j; i++, j--) {
            if (s[i] != s[j]) {
                return isOk(s, i, j - 1) || isOk(s, i + 1, j);
            }
        }
        return true;
    }
};